src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -2,18 +2,127 @@
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+
+    private Node<K, V>[] table;
+    private int size;
+    private int capacity;
+    private final float loadFactor;
+    private int threshold;
+
+    private V nullKeyValue = null;
+    private boolean nullKeyPresent = false;
```

Comment:
Consider removing these fields and handling the null key scenario within your existing data structure. These fields seem to be a workaround for a specific case that could be integrated into the main logic.